### PR TITLE
docs: add support for https

### PR DESCRIPTION
**Pull request by tisfeng**
close https://github.com/aaamoon/copilot-gpt4-service/issues/99

### CODE REVIEW

**Review by Geniucker — APPROVED** [review]

**Comment by funnyzak** [comment]
How to set up HTTPS, which is not a functional explanation of this service, and it is recommended for local use. Should it be placed in Frequently Asked Questions?

**Comment by tisfeng** [comment]
Support for HTTPS is necessary, the current macOS system API only supports HTTPS requests by default, and non-HTTPS requests will fail unless the app explicitly bypasses the HTTPS security settings. There are also apps that are set to only support HTTPS, such as `OpenCat` https://github.com/aaamoon/copilot-gpt4-service/issues/99 .

Isn't it easier and simpler to get users to support HTTPS just by adding a caddy instruction than to manually configure a certificate at https://github.com/aaamoon/copilot-gpt4-service/pull/103? This approach is much more user-friendly for beginners.

If you think it's taking up too much space, we can add a `<details>` tag to turn it off by default.

>  it is recommended for local use

Why? Are there any problems with deploying it on a server? The server is more stable, and it's also easier for users to use it at home or in the company, as well as working better with services such as [one-api](https://github.com/songquanpeng/one-api).

> Should it be placed in Frequently Asked Questions?

I don't think this is an FAQ, it's a deployment configuration option. Of course if you think it needs to be in the FAQ that's fine, it's up to you.

**Comment by tisfeng** [comment]
I added a `<details>` tag to wrap https docs.

If you guys have a better idea, you can place it in another location as well.

image

**Comment by tisfeng** [comment]
I understand that, and using Copilot's API directly may not be in line with their company policy, but it has very little to do with using it locally or on a server, they're both running copilot-gpt4-service, except the server uses a different IP than the local one.

But I don't think that's a small enough difference to qualify as a risk. Before using copilot-gpt4-service, I used Github Copilot with a proxy every day, often switching back and forth between several IPs, and I think that Github's risk control was pretty good, and they didn't block my account just because I changed a few IPs (not dozens).

I think as long as we try to simulate the same request as Github Copilot, and optimize the code, e.g. https://github.com/aaamoon/copilot-gpt4-service/pull/188, we can use this service relatively safely.

**Review by funnyzak — APPROVED** [review]

File: README_CN.md [hunk 1]
```diff
@@ -108,6 +111,61 @@ helm repo add aaamoon https://charts.kii.la && helm repo update # 源由 github
...
+sudo apt install caddy
+```
+
+### 配置 Caddy
+
+```bash
+sudo vi /etc/caddy/Caddyfile
+```
+
+假如你准备使用的域名为 `your.domain.com`，请先进行 DNS 解析，将域名解析到服务器 IP 地址，然后在 Caddyfile 中添加以下内容：
```
Comment (Geniucker) [inline README_CN.md:140]: Is if necessary to mention `80` and `443` ports should be allowed in firewall in advance or Caddy can't get certification through ACME

Comment (tisfeng) [reply README_CN.md:140]: Sure, I will add this.

**Review by Geniucker — APPROVED** [review]

**Comment by missuo** [comment]
Since Caddy has been added, let's also add Nginx.

**Comment by tisfeng** [comment]
Sorry, I'm not familiar with Nginx, I can't help 😔

**Pull request merged**